ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpDataHelper.java

```diff
@@ -0,0 +1,630 @@
+        }
+        randomAccessFile.seek(++tempCurrentOffset);
+        currentOffset = tempCurrentOffset;
+        lastBlobFailed = true;
+      } catch (EOFException e) {
+        e.printStackTrace();
+        logOutput("EOFException thrown at " + randomAccessFile.getChannel().position() + ", Cause :" + e.getCause()
+            + ", Msg :" + e.getMessage(), LogLevel.INFO);
+        throw (e);
+      } catch (Exception e) {
+        if (!lastBlobFailed) {
+          e.printStackTrace();
+          logOutput(
+              "Unknown exception thrown " + e.getMessage() + "\nTrying from next offset " + (tempCurrentOffset + 1)
+                  + ", Cause " + e.getCause() + ", Msg :" + e.getMessage(), LogLevel.INFO);
+        }
+        randomAccessFile.seek(++tempCurrentOffset);
+        currentOffset = tempCurrentOffset;
+        lastBlobFailed = true;
+      }
+    }
+    logOutput("Dumped until offset " + currentOffset, LogLevel.INFO);
+  }
+
+  /**
+   * Holds information about a blob record in the log
+   */
+  class BlobRecordInfo {
+    String messageheader = null;
+    BlobId blobId = null;
+    String blobProperty = null;
+    String usermetadata = null;
+    String blobDataOutput = null;
+    String deleteMsg = null;
+    boolean isDeleted;
+    boolean isExpired;
+    long timeToLiveInSeconds;
+    int totalRecordSize;
+
+    public BlobRecordInfo(String messageheader, BlobId blobId, String blobProperty, String usermetadata,
+        String blobDataOutput, String deleteMsg, boolean isDeleted, boolean isExpired, long timeToLiveInSeconds,
+        int totalRecordSize) {
+      this.messageheader = messageheader;
+      this.blobId = blobId;
+      this.blobProperty = blobProperty;
+      this.usermetadata = usermetadata;
+      this.blobDataOutput = blobDataOutput;
+      this.deleteMsg = deleteMsg;
+      this.isDeleted = isDeleted;
+      this.isExpired = isExpired;
+      this.timeToLiveInSeconds = timeToLiveInSeconds;
+      this.totalRecordSize = totalRecordSize;
+    }
+  }
+
+  /**
+   * Fetches one blob record from the log
+   * @param randomAccessFile {@link RandomAccessFile} referring to the log file
+   * @param currentOffset the offset at which to read the record from
+   * @return the {@link BlobRecordInfo} containing the blob record info
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  private BlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFile, long currentOffset)
+      throws IOException, MessageFormatException {
+    String messageheader = null;
+    BlobId blobId = null;
+    String blobProperty = null;
+    String usermetadata = null;
+    String blobDataOutput = null;
+    String deleteMsg = null;
+    boolean isDeleted = false;
+    boolean isExpired = false;
+    long timeToLiveInSeconds = -1;
+    int totalRecordSize = 0;
+    short version = randomAccessFile.readShort();
+    if (version == 1) {
```

Comment:
this is header version. We have just one header version for now. Check Message_Header_Version_V1 in MessageFormatRecord